dotCMS/src/main/java/com/dotcms/rest/api/v1/contenttype/ContentTypeForm.java

```diff
@@ -0,0 +1,129 @@
+import com.dotcms.repackage.com.google.common.annotations.VisibleForTesting;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.json.JSONArray;
+import com.dotmarketing.util.json.JSONException;
+import com.dotmarketing.util.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * {@link ContentTypeResource}'s form
+ */
+@JsonDeserialize(using = ContentTypeForm.ContentTypeFormDeserialize.class)
+public class ContentTypeForm  {
+
+    private List<ContentTypeFormEntry> entries;
+
+    public ContentTypeForm(List<ContentTypeFormEntry> entries) {
+        this.entries = entries;
+    }
+
+    public Iterable<ContentTypeFormEntry> getIterable() {
+        return entries;
+    }
+
+    public ContentType getContentType() {
+        return entries.get(0).contentType;
+    }
+
+    public List<String> getWorkflowsIds() {
+        return entries.get(0).workflowsIds;
+    }
+
+    public static final class ContentTypeFormDeserialize extends JsonDeserializer<ContentTypeForm> {
+
+        public static final String WORKFLOW_ATTRIBUTE_NAME = "workflow";
+
+        @Override
+        public ContentTypeForm deserialize(JsonParser jsonParser, DeserializationContext deserializationContext)
+                throws IOException {
+
+            String json = jsonParser.readValueAsTree().toString();
+            return buildForm(json);
+        }
+
+        @VisibleForTesting
+        public ContentTypeForm buildForm(String json) {
+            final List<ContentType> typesToSave = new JsonContentTypeTransformer(json).asList();
+            final List<List<String>> workflows = getWorkflowIdsFromJson(json);
+
+            List<ContentTypeFormEntry> entries = getContentTypeFormEntries(typesToSave, workflows);
+
+            return new ContentTypeForm(entries);
+        }
+
+        private List<ContentTypeFormEntry> getContentTypeFormEntries(List<ContentType> typesToSave,
+                                                                     List<List<String>> workflows) {
+
+            List<ContentTypeFormEntry> entries = new ArrayList<>();
+
+            for (int i = 0; i < workflows.size(); i++) {
+                List<String> worflows = workflows.get(i);
+                ContentType contentType = typesToSave.get(i);
+
+                ContentTypeFormEntry entry = new ContentTypeFormEntry(contentType, worflows);
+                entries.add(entry);
+            }
+            return entries;
+        }
+
+        private static List<List<String>> getWorkflowIdsFromJson(String json) {
+            final List<List<String>> workflows = new ArrayList<>();
+
+            try {
+                JSONArray jarr = new JSONArray(json);
+
+                for (int i = 0; i < jarr.size(); i++) {
+                    final JSONObject fieldJsonObject = (JSONObject) jarr.get(i);
+
+                    if (fieldJsonObject.has("workflow")){
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use already-defined constant 'WORKFLOW_ATTRIBUTE_NAME' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)